src/Checkpointer.jl

```diff
@@ -146,6 +146,9 @@ for the entire cache structure. This function addresses three key issues:
 3. **Selective saving**: Only the parts of the cache needed for restoration are
    saved.
 
+4. **Recreated views**: Views are preserved by recreating the views with the
+   parent arrays on CPU.
```

Comment:
```suggestion
   parent arrays on CPU. Otherwise, the arrays are unnecessarily duplicated when `adapt` is called since checks using `objectid` fail for views, resulting in a larger saved cache on GPU.
```